src/splitobs.jl

```diff
@@ -0,0 +1,88 @@
+"""
+    splitobs(n::Int; at) -> Tuple
+
+Pre-compute the indices for two disjoint subsets and return them
+as a tuple of two ranges. The first range will span the first
+`at` fraction of possible indices, while the second range will
+cover the rest. These indices are applicable to any data
+container of size `n`.
+
+```julia
+julia> splitobs(100, at = 0.7)
+(1:70, 71:100)
+```
+"""
+splitobs(n::Int; at) = _splitobs(n, at)
+
+_splitobs(n::Int, at::Tuple{}) = (1:n,)
+
+function _splitobs(n::Int, at::Number)
+    0 <= at <= 1 || throw(ArgumentError("the parameter \"at\" must be in interval (0, 1)"))
```

Comment:
It would be nice to allow `at` to be an integer for exact splits. Thoughts?

Reply:
yes. We can dispatch on the type of `at`, I'll add it.

Reply:
done